File: PythonCodeSnippets/snip_frequency_noisy_cosine.py
import numpy as np
import matplotlib.pyplot as plt

N = 1024
A = 4  # Cosine amplitude of A Volts
Fs = 8000
Ts = 1 / Fs  # Sampling frequency (Hz) and period (s)
f0 = 915  # Cosine frequency in Hz
noise_power = 16  # Noise power in Watts

# Generate time vector
t = np.arange(0, N * Ts, Ts)

# Generate cosine with AWGN
noise = np.sqrt(noise_power) * np.random.randn(N)
x = A * np.cos(2 * np.pi * f0 * t) + noise

# Periodogram
f, P = plt.psd(x, NFFT=N, Fs=Fs, scale_by_freq=False)

# Plot periodogram on a dB scale
plt.plot(f, 10 * np.log10(P))
plt.xlabel('Frequency (Hz)')
plt.ylabel('Power/Frequency (dB/Hz)')
plt.show()

File: MatlabOctaveCodeSnippets/snip_frequency_noisy_cosine.m
N=1024; A=4; %# of samples and cosine amplitude of A Volts
Fs=8000; Ts=1/Fs; %sampling frequency (Hz) and period (s)
f0=915; %cosine frequency in Hz
noisePower=16; %noise power in Watts
noise=sqrt(noisePower)*randn(1,N);
t=0:Ts:(N-1)*Ts; %N time instants separated by Ts
x=A*cos(2*pi*f0*t) + noise;%generate cosine with AWGN
[P,f]=periodogram(x,[],N,Fs); %periodogram (using the default window)
plot(f,10*log10(P)); %plot in dB scale